scripts.js

```diff
@@ -14,10 +16,112 @@ function createTileObject(name, top, left) {
   newTile.innerText = name;
   newTile.style.top = top + 'px';
   newTile.style.left = left + 'px';
+  newTile.id = Math.floor(Math.random() * 1000);
+  newTile.addEventListener('click', function () {
+    moveTile(this.id, this.parentElement);
+  });
+  newTile.draggable = 'true';
+  newTile.addEventListener('dragstart', function (ev) {
+    ev.dataTransfer.setData('div', ev.target.id);
+  });
+
+  tilesFirstContainer.push(newTile);
 
   return newTile;
 }
 
-function addTile() {}
+function addTile() {
+  const input = document.getElementById('tile-name').value;
+  const randomTop = Math.floor(Math.random() * (400 - 40));
+  const randomLeft = Math.floor(Math.random() * (500 - 120));
+  const tile1 = createTileObject(input, randomTop, randomLeft);
+  container1.appendChild(tile1);
+}
+
+function sortOneContainer(container, tilesToSort) {
+  tilesToSort.sort((a, b) => a.innerText.localeCompare(b.innerText));
+
+  let currentY = -40;
+  let currentX = 0;
+  let heightToAdd = 10;
+  let marginLeft = 10;
+  let marginTop = 0;
+
+  // 120 - width of tile, 30 - total padding on line
+  const maxElementsOnLine = parseInt(container.offsetWidth / (120 + 30));
+  //400 - minHeight of container, 60 - sum of: padding (20 - top+bottom) and height of tile (40)
+  const maxElementsInCont = parseInt((400 / 60) * maxElementsOnLine);
+
+  container.innerHTML = '';
+
+  for (let i = 0; i < tilesToSort.length; i++) {
+    if (i % maxElementsOnLine === 0) {
+      currentY = currentY + 40;
+      currentX = 0;
+      marginLeft = (container.offsetWidth - maxElementsOnLine * (120 + 10)) / 2;
+
+      marginTop = marginTop + 10;
+
+      if (i >= maxElementsInCont) {
+        container.style.minHeight = 400 + heightToAdd + 'px';
+
+        heightToAdd = heightToAdd + 50;
+      }
+    } else {
+      currentX = currentX + 120;
+    }
+
+    tilesToSort[i].style.top = currentY + 'px';
+    tilesToSort[i].style.left = currentX + 'px';
+    tilesToSort[i].style.marginLeft = marginLeft + 'px';
+    tilesToSort[i].style.marginTop = marginTop + 'px';
+
+    marginLeft = marginLeft + 10;
 
-function sortTiles() {}
+    container.appendChild(tilesToSort[i]);
+  }
+}
+
+function sortTiles() {
+  sortOneContainer(container1, tilesFirstContainer);
+  sortOneContainer(container2, tilesSecondContainer);
+}
+
+function moveTile(id, container) {
+  if (container == container1) {
+    const tileToMove = tilesFirstContainer.find((tile) => tile.id === id);
```

Comment:
i think the "if" section is very similar to the "else" section. Think about how you can extract the functionality in a method so you don't have repeating code.